lms/djangoapps/verify_student/views.py

```diff
@@ -270,35 +271,41 @@ def get(
         if redirect_url:
             return redirect(redirect_url)
 
-        # Check that the course has an unexpired verified mode
-        course_mode, expired_course_mode = self._get_verified_modes_for_course(course_key)
-
-        if course_mode is not None:
-            log.info(
-                u"Entering verified workflow for user '%s', course '%s', with current step '%s'.",
-                request.user.id, course_id, current_step
-            )
-        elif expired_course_mode is not None:
-            # Check if there is an *expired* verified course mode;
-            # if so, we should show a message explaining that the verification
-            # deadline has passed.
-            log.info(u"Verification deadline for '%s' has passed.", course_id)
-            context = {
-                'course': course,
-                'deadline': (
-                    get_default_time_display(expired_course_mode.expiration_datetime)
-                    if expired_course_mode.expiration_datetime else ""
+        # Check that the course has an unexpired no-id-professional mode
+        no_id_prof_mode = CourseMode.mode_for_course(course_key, CourseMode.NO_ID_PROFESSIONAL_MODE[0])
```

Comment:
I'm concerned with this approach: it requires an additional database query and requires this view to make too many assumptions about available course modes.

I'd prefer to retrieve all the modes for the course, then check whether the modes are (a) verified and/or (b) paid.  You can use this information to decide which steps to include without hard-coding the new course mode (No-ID prof ed).